### PR TITLE
Dev bish22ah

### PR DESCRIPTION
**Pull request by bish22ah**
1. Prevent NotReady state to Clients
2. Dropbox class updates
3. Custom exception AutoHCKError class
4. Better logging across multiple classes
5. Added exceptions to handle failed VirtHCK class runs
6. Updating return_when_client_up mechanism to use Thread's built in timeout
7. Studio class added StudioConnectionError exception class
8. Retry if VirtHCK orchestration failed
9. Disable TERM signal handling in case of a fatal error

### CODE REVIEW

File: lib/client.rb [hunk 1]
```diff
@@ -208,6 +214,13 @@ def alive?
     false
   end
 
+  def prevent_not_ready
```
Comment (YanVugenfirer) [inline lib/client.rb]: I would rename to "reset_to_ready_state" or something similar.

File: lib/tests.rb [hunk 2]
```diff
@@ -115,6 +115,11 @@ def handle_finished_tests(tests)
     print_tests_stats
   end
 
+  def prevent_clients_not_ready
```
Comment (YanVugenfirer) [inline lib/tests.rb]: Same here: reset_clients_to_ready_state or clients_reset_to_ready_state or if this is client class already just reset_to_ready_state

File: bin/auto_hck [hunk 3]
```diff
@@ -53,6 +53,7 @@ Signal.trap('TERM') do
 end
 
 Thread.abort_on_exception = true
+Thread.report_on_exception = false
```
Comment (YanVugenfirer) [inline bin/auto_hck]: "report_exception?" probably better

Comment (bish22ah) [reply bin/auto_hck]: This only prevents the thread exception to be reported to stderr, the exception will still be raised from the thread to the main thread.

File: lib/client.rb (hunk 1 above)
Comment (bish22ah) [reply lib/client.rb]: On it...

File: lib/tests.rb (hunk 2 above)
Comment (bish22ah) [reply lib/tests.rb]: On it...

**Pull request merged**